### PR TITLE
Cleanup ASCIIEncoding tests

### PR DESCRIPTION
**Pull request by hughbe**
- Rename tests to follow guidelines, instead of "PosTest1" and
"NegTest2"
- Take advantage of xunit
- Shorten tests by using quicker and newer code

### CODE REVIEW

File: src/System.Text.Encoding/tests/ASCIIEncoding/ASCIIEncodingGetByteCount.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                Assert.Equal(expected, new ASCIIEncoding().GetByteCount(chars));
+                Assert.Equal(expected, new ASCIIEncoding().GetByteCount(charArray));
+            }
+            Assert.Equal(expected, new ASCIIEncoding().GetByteCount(charArray, index, count));
+        }
+
+        [Fact]
+        public void GetByteCount_Invalid()
+        {
+            // Chars is null
```
Comment (stephentoub) [inline src/System.Text.Encoding/tests/ASCIIEncoding/ASCIIEncodingGetByteCount.cs]: For all of these, can you add in the argument name to the Throws validation?

File: src/System.Text.Encoding/tests/ASCIIEncoding/ASCIIEncodingGetBytes.cs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                    ++currentByteIndex;
+                }
+            }
+            Assert.Equal(currentByteIndex - byteIndex, result);
+        }
+
+        [Fact]
+        public void GetBytes_Invalid()
+        {
+            // Source is null
```
Comment (stephentoub) [inline src/System.Text.Encoding/tests/ASCIIEncoding/ASCIIEncodingGetBytes.cs]: Same arg exception names comment as above.

File: src/System.Text.Encoding/tests/DecoderFallbackException/DecoderFallbackExceptionTests.cs [hunk 3]
```diff
@@ -2,67 +2,57 @@
...
-            DecoderFallbackException ex = new DecoderFallbackException();
-            Assert.Null(ex.BytesUnknown);
-            Assert.Equal(default(int), ex.Index);
-            Assert.Null(ex.StackTrace);
-            Assert.Null(ex.InnerException);
-            Assert.Equal(0, ex.Data.Count);
-            ArgumentException arg = new ArgumentException();
+            DecoderFallbackException decoderFallbackException = new DecoderFallbackException();
+            Assert.Null(decoderFallbackException.BytesUnknown);
+            Assert.Equal(default(int), decoderFallbackException.Index);
```
Comment (stephentoub) [inline src/System.Text.Encoding/tests/DecoderFallbackException/DecoderFallbackExceptionTests.cs]: Nit: 0?

File: src/System.Text.Encoding/tests/DecoderFallbackException/DecoderFallbackExceptionTests.cs [hunk 4]
```diff
@@ -2,67 +2,57 @@
...
-            DecoderFallbackException ex = new DecoderFallbackException(message);
-            Assert.Null(ex.BytesUnknown);
-            Assert.Equal(default(int), ex.Index);
-            Assert.Null(ex.StackTrace);
-            Assert.Null(ex.InnerException);
-            Assert.Equal(0, ex.Data.Count);
-            Assert.Equal(ex.Message, message);
+            DecoderFallbackException decoderFallbackException = new DecoderFallbackException(message);
+            Assert.Null(decoderFallbackException.BytesUnknown);
+            Assert.Equal(default(int), decoderFallbackException.Index);
```
Comment (stephentoub) [inline src/System.Text.Encoding/tests/DecoderFallbackException/DecoderFallbackExceptionTests.cs]: Nit: 0?

File: src/System.Text.Encoding/tests/EncoderFallbackException/EncoderFallbackExceptionTests.cs [hunk 5]
```diff
@@ -2,85 +2,69 @@
...
-            EncoderFallbackException ex = new EncoderFallbackException();
-            Assert.Equal(default(char), ex.CharUnknown);
-            Assert.Equal(default(char), ex.CharUnknownHigh);
-            Assert.Equal(default(char), ex.CharUnknownLow);
-            Assert.Equal(default(int), ex.Index);
+            EncoderFallbackException encoderFallbackException = new EncoderFallbackException();
+            Assert.Equal(default(char), encoderFallbackException.CharUnknown);
+            Assert.Equal(default(char), encoderFallbackException.CharUnknownHigh);
+            Assert.Equal(default(char), encoderFallbackException.CharUnknownLow);
+            Assert.Equal(default(int), encoderFallbackException.Index);
```
Comment (stephentoub) [inline src/System.Text.Encoding/tests/EncoderFallbackException/EncoderFallbackExceptionTests.cs]: Same nits :) I'll stop mentioning it.

**Comment by stephentoub** [comment]
A few nits, otherwise LGTM. Thanks for cleaning these up.

**Comment by hughbe** [comment]
Thanks for the comments.
I pushed a PR feedback commit solving those nits and also adding two missing tests for exception throwing

**Comment by stephentoub** [comment]
Thanks! LGTM

**Pull request merged**